### ISSUE TITLE
[Bug]: delete entity does not work

### ISSUE THREAD

**Issue by DrBamba** [milvus-io/milvus#15622]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.0
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus 2.0.0
- OS(Ubuntu or CentOS): ubuntu
- CPU/Memory: 
- GPU: 
- Others: deployed on openshift
```

### Current Behavior

When I delete an entity using the python SDK, it doesn't show an error but it does not delete the entity. Even if I wait for a couple of minutes or using strong consistency.
However if I am using Attu and deleting the entity, it is deleted immediately

### Expected Behavior

When I delete an entity using the python SDK it should be deleted

### Steps To Reproduce

1. Create a collection
2. Load collection
3. Insert vectors
4. Query collection
5. Delete a vector
6. Query collection

### Anything else?

Data node logs: 
```
maxprocs: Updating GOMAXPROCS=2: determined from CPU quota

    __  _________ _   ____  ______
   /  |/  /  _/ /| | / / / / / __/
  / /|_/ // // /_| |/ / /_/ /\ \
 /_/  /_/___/____/___/\____/___/

Welcome to use Milvus!
Version:   v2.0.0
Built:
GitCommit: 6336e232
GoVersion: go version go1.15.2 linux/amd64

open pid file: /run/milvus/datanode.pid
open pid file: /run/milvus/datanode.pid
[DEBUG] [base_table.go:65] ["config directory"] [configDir=/milvus/configs/]
[DEBUG] [global_param.go:973] ["init cacheSize"] ["cacheSize (GB)"=0]
[DEBUG] [global_param.go:1042] ["query node init gracefulTime"] [gracefulTime=0]
[DEBUG] [base_table.go:65] ["config directory"] [configDir=/milvus/configs/]
[DEBUG] [global_param.go:1523] [initServerMaxSendSize] [role=datanode] [grpc.serverMaxSendSize=2147483647]
[DEBUG] [global_param.go:1546] [initServerMaxRecvSize] [role=datanode] [grpc.serverMaxSendSize=2147483647]
[DEBUG] [service.go:238] ["DataNode address"] [address=10.130.5.234:21124]
[DEBUG] [service.go:247] ["Init root coord client ..."]
[DEBUG] [session_util.go:86] ["Session try connect to etcd"]
[DEBUG] [session_util.go:101] ["Session connect to etcd success"]
[DEBUG] [base_table.go:65] ["config directory"] [configDir=/milvus/configs/]
[DEBUG] [global_param.go:1591] [initClientMaxSendSize] [role=rootcoord] [grpc.clientMaxSendSize=104857600]
[DEBUG] [global_param.go:1614] [initClientMaxRecvSize] [role=rootcoord] [grpc.clientMaxSendSize=104857600]
[DEBUG] [session_util.go:294] ["SessionUtil GetSessions "] [prefix=rootcoord] [key=rootcoord] [address=10.130.8.108:53100]
[DEBUG] [client.go:91] ["RootCoordClient GetSessions success"]
[DEBUG] [service.go:266] ["DataNode rootcoord is ready"]
[DEBUG] [service.go:274] ["DataNode Init data service client ..."]
[DEBUG] [session_util.go:86] ["Session try to connect to etcd"]
[DEBUG] [session_util.go:101] ["Session connect to etcd success"]
[DEBUG] [base_table.go:65] ["config directory"] [configDir=/milvus/configs/]
[DEBUG] [global_param.go:1591] [initClientMaxSendSize] [role=datacoord] [grpc.clientMaxSendSize=104857600]
[DEBUG] [global_param.go:1614] [initClientMaxRecvSize] [role=datacoord] [grpc.clientMaxSendSize=104857600]
[DEBUG] [session_util.go:294] ["SessionUtil GetSessions "] [prefix=datacoord] [key=datacoord] [address=10.132.5.50:13333]
[DEBUG] [service.go:293] ["DataNode dataCoord is ready"]
[DEBUG] [data_node.go:209] ["DataNode Init"] [TimeTickChannelName=by-dev-datacoord-timetick-channel]
[DEBUG] [session_util.go:86] ["Session try connect to etcd"]
[DEBUG] [session_util.go:101] ["Session connect to etcd success"]
[DEBUG] [session_util.go:175] ["Session get serverID success"] [key=id] [serverId=17]
[DEBUG] [base_table.go:479] ["Set logger "] [id=17] [role=datanode]
[INFO] [base_table.go:466] ["Set log file to "] [path=/milvus/logs/datanode-17.log]
level=info msg="[Connecting to broker]" remote_addr="pulsar://milvus-pulsar.project.svc:6650"
level=info msg="[TCP connection established]" local_addr="10.130.5.234:53628" remote_addr="pulsar://milvus-pulsar.project.svc:6650"
level=info msg="[Connection is ready]" local_addr="10.130.5.234:53628" remote_addr="pulsar://milvus-pulsar.project.svc:6650"
level=info msg="[Connecting to broker]" remote_addr="pulsar://milvus-pulsar.project.svc:6650"
level=info msg="[TCP connection established]" local_addr="10.130.5.234:53628" remote_addr="pulsar://milvus-pulsar.project.svc:6650"
level=info msg="[Connection is ready]" local_addr="10.130.5.234:53628" remote_addr="pulsar://milvus-pulsar.project.svc:6650"
level=info msg="[Connected consumer]" consumerID=1 name=bznsj subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_2"
level=info msg="[Created consumer]" consumerID=1 name=bznsj subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_2"
level=info msg="Broker notification of Closed consumer: 1" local_addr="10.130.5.234:53630" remote_addr="pulsar://milvus-pulsar.project.svc:6650"
level=info msg="[Reconnecting to broker in 115.465373ms]"]" consumerID=1 name=bznsj subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_2"
level=info msg="[Connected consumer]" consumerID=1 name=bznsj subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_2"
level=info msg="[Reconnected consumer to broker]" consumerID=1 name=bznsj subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_2"
level=info msg="[Created producer]" cnx="10.130.5.234:53630 -> 172.31.191.12:6650" producerID=1 producer_name=standalone-0-42 topic="persistent://public/default/by-dev-rootcoord-delta_2"
level=info msg="[Created producer]" cnx="10.130.5.234:53630 -> 172.31.191.12:6650" producerID=2 producer_name=standalone-0-43 topic="persistent://public/default/by-dev-rootcoord-timetick-channel"
level=info msg="[Connected consumer]" consumerID=2 name=opwgt subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_3"
level=info msg="[Created consumer]" consumerID=2 name=opwgt subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_3"
level=info msg="Broker notification of Closed consumer: 2" local_addr="10.130.5.234:53630" remote_addr="pulsar://milvus-pulsar.project.svc:6650"
level=info msg="[Reconnecting to broker in 107.365419ms]"]" consumerID=1 name=opwgt subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_3"
level=info msg="[Connected consumer]" consumerID=2 name=opwgt subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_3"
level=info msg="[Reconnected consumer to broker]" consumerID=2 name=opwgt subscription=by-dev-dataNode-17-431257613034848257 topic="persistent://public/default/by-dev-rootcoord-dml_3"
level=info msg="[Created producer]" cnx="10.130.5.234:53630 -> 172.31.191.12:6650" producerID=3 producer_name=standalone-0-44 topic="persistent://public/default/by-dev-rootcoord-delta_3"
level=info msg="[Created producer]" cnx="10.130.5.234:53630 -> 172.31.191.12:6650" producerID=4 producer_name=standalone-0-45 topic="persistent://public/default/by-dev-rootcoord-timetick-channel"
panic: execution failed

goroutine 2842 [running]:
github.com/milvus-io/internal/datanode.flushNotifyFunc.func1(0xc0008ef090)
	/go/src/github.com/milvus-io/internal/datanode/flush_manager.go:693 +0x1645
github.com/milvus-io/internal/datanode.(*flushTaskRunner).waitFinish(0xc000282f00, 0xc0006273b0, 0xc000d3a4a0)
	/go/src/github.com/milvus-io/internal/datanode/flush_task.go:194 +0xbd
created by github.com/milvus-io/internal/datanode.(*flushTaskRunner).init.func1
	/go/src/github.com/milvus-io/internal/datanode/flush_task.go:119 +0xb0
```

EDIT: 
Dimensions of vectors is 512
Flat index
Delete only one vector
Tried to delete using `id in [<vector_id>]` for example: `id in [12345]`
Query to check if exists with `id == <vector_id>`
Strong consistency didn't seem to have an effect

EDIT2:
Extended logs to the full log file

EDIT3:
Cluster is deployed on openshift (including pulsar and etcd) using S3 instead of minio

**Comment by yanliang567** [comment]
@jingkl could you please help to reproduce this issue? I believe there are some tests for this scenarios.
/assign @jingkl 
/unassign

**Comment by DrBamba** [comment]
> I will try to reproduce the problem as much as possible. If possible need to add relevant parameters like dimension of vector, relevant parameters of query, delete only one vector? Thanks @DrBamba

Dimensions of vectors is 512
Flat index
Delete only one vector
Tried to delete using `id in [<vector_id>]` for example: `id in [12345]`
Query to check if exists with `id == <vector_id>`
Strong consistency didn't seem to have an effect

**Comment by jingkl** [comment]
I tested this with the following script and found that the consistency setting of "strong" returned a successful deletion
```
from time import sleep
from pymilvus import connections, Collection, FieldSchema, CollectionSchema, DataType
import random
connections.connect()
fields = [
    FieldSchema("film_id", DataType.INT64, is_primary=True),
    FieldSchema("films", dtype=DataType.FLOAT_VECTOR, dim=512)
]
description="This is an example text description."
schema = CollectionSchema(fields=fields, description=description)
collection = Collection(name="test_collection_5", schema=schema, consistency_level="Strong" )
collection.load()
data = [
    [i for i in range(100)],
    [[random.random() for _ in range(512)] for _ in range(100)],
]
collection.insert(data)
print(collection.num_entities)
index_params = {
        "metric_type":"L2",
        "index_type":"IVF_FLAT",
        "params":{"nlist":1024}
    }
collection.create_index(
    field_name="films", 
    index_params=index_params
    )

expr = "film_id in [ 0 ]"
res = collection.query(expr, output_fields=["films"])
print(f"- Query results: {res}")

expr = "film_id in [ 0 ]"
res = collection.delete(expr)
print(f"- Deleted entities: {res}")

# # sleep(30)

expr = expr = "film_id in [ 0]"
res = collection.query(expr, output_fields=["films"])
print(f"-  Query if ID exists results: {res}")
```
<img width="1436" alt="截屏2022-02-18 16 35 13" src="https://user-images.githubusercontent.com/34296482/154647201-4b8a93ab-2ae4-4d16-834a-2fd1793a8cee.png">
As shown in the picture, IDs that have been deleted cannot be looked up
@DrBamba

**Comment by DrBamba** [comment]
@jingkl thank you for the response, the only differences I see are those:
1. In consistency level I used the type, instead of hardcoded "Strong" (should not matter cause the value is the same)
2. I used a flat index, cause data accuracy is important to me (didn't create an index, which is the same to my understanding)

Other then that it seems to be similar, the only extra information I have is that I deployed the cluster on an openshift environment, using two nodes of each and one for each coord.
The data nodes start to restart non stop after I try to perform the operations I described, printing the logs I attached to the issue

**Comment by jingkl** [comment]
> @jingkl thank you for the response, the only differences I see are those:
> 
> 1. In consistency level I used the type, instead of hardcoded "Strong" (should not matter cause the value is the same)
> 2. I used a flat index, cause data accuracy is important to me (didn't create an index, which is the same to my understanding)
> 
> Other then that it seems to be similar, the only extra information I have is that I deployed the cluster on an openshift environment, using two nodes of each and one for each coord. The data nodes start to restart non stop after I try to perform the operations I described, printing the logs I attached to the issue

What does the use of "type" in the consistency level mean? Can you upload the full datanode log? As a zip archive
@DrBamba

**Comment by DrBamba** [comment]
@jingkl I edited the issue to include the full data node log as you requested :)

**Comment by jingkl** [comment]
> I edited the issue to include the full data node log as you requested :)

I will try again, thanks

**Comment by DrBamba** [comment]
I redeployed pulsar with a clean image 2.7.3, also I noticed the milvus configuration file was not updated so I replaced it with the new one, bit still the exact same error
Any progress on your side? @jingkl

It seems that unless I try to delete something everything works as it should, by the logs it seems that the error occurs when flushing

**Comment by jingkl** [comment]
> I redeployed pulsar with a clean image 2.7.3, also I noticed the milvus configuration file was not updated so I replaced it with the new one, bit still the exact same error Any progress on your side? @jingkl
> 
> It seems that unless I try to delete something everything works as it should, by the logs it seems that the error occurs when flushing
> 
> Also I should note the cluster is deployed on an openshift environment

I am still missing relevant logs such as datacoord and pulsar to help me analyse further?

**Comment by jingkl** [comment]
> I no longer experience this issue, I am not sure as to what resolved the issue but it seemed that it might have been originated in errors with the S3 I was using causing errors when milvus tries to flush.
> 
> Should maybe have a way to validate that the S3/minio is configured correctly and show an indicative error otherwise

Thanks for the advice, can you tell me how you configured the S3 before? Can you tell me the relevant configuration?
@DrBamba

**Comment by xiaofan-luan** [comment]
@jeff-zimmerman 
this issue has been fixed on 2.2.13, please open a new issue if you still find any corner case on 2.2.13 or above
